private.go

```diff
@@ -6,28 +6,36 @@ import (
 	ma "github.com/multiformats/go-multiaddr"
 )
 
-// Private4 and Private6 are well-known private networks
-var Private4, Private6 []*net.IPNet
-var privateCIDR4 = []string{
+// Private4, Private6, Local4, and Local6 are well-known private networks
+var Private4, Private6, Local4, Local6 []*net.IPNet
+
+var localCIDR4 = []string{
 	// localhost
 	"127.0.0.0/8",
+	// link local
+	"169.254.0.0/16",
```

Comment:
I think it’s ok moving link locals here. I struggle to find a use case where these would be useful, but maybe I’m missing something, @stebalien?